universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -53,4 +57,22 @@ public QuestionDefinition update(QuestionDefinition definition) {
                     .map(question -> question.getQuestionDefinition())
                     .collect(ImmutableList.toImmutableList()));
   }
+
+  private boolean isValid(QuestionDefinition definition) {
+    String newPath = definition.getPath();
+    if (!isValidPathPattern(newPath)) {
+      return false;
+    }
+    boolean hasConflict =
+        questionRepository.lookupPathConflict(newPath).toCompletableFuture().join().booleanValue();
+    return !hasConflict;
+  }
+
+  private boolean isValidPathPattern(String path) {
+    try {
+      return URLEncoder.encode(path, StandardCharsets.UTF_8.name()).equals(path);
+    } catch (Exception e) {
```

Comment:
Suggestion (blocking): never catch the generic `Exception` since it could hide bugs that have nothing to do with the exception you're expecting to catch here. Instead, catch the exception you expect `URLEncoder.encode` to throw on bad input specifically (you can use a test to discover what it is).

Reply:
It turns out that this is not necessary. The only exception is thrown when the `Charset` name is unrecognized. I changed it to the static field.